### PR TITLE
Parse scope with HTTPConnection instead of Request in SessionMiddleware

### LINKED ISSUE #483: Session cookie triggers failure on websocket connection upgrade
Hi,

When enabling the session cookie on an app, if a websocket connection upgrade is made, then we seem to fall into a an assertion failuire:

```
Traceback (most recent call last):
  File ".venv/lib/python3.7/site-packages/uvicorn/protocols/websockets/websockets_impl.py", line 146, in run_asgi
    asgi = self.app(self.scope)
  File ".venv/lib/python3.7/site-packages/starlette/applications.py", line 134, in __call__
    return self.error_middleware(scope)
  File ".venv/lib/python3.7/site-packages/starlette/middleware/errors.py", line 88, in __call__
    return self.app(scope)
  File ".venv/lib/python3.7/site-packages/uvicorn/middleware/proxy_headers.py", line 37, in __call__
    return self.app(scope)
  File ".venv/lib/python3.7/site-packages/starlette/middleware/base.py", line 22, in __call__
    return self.app(scope)
  File ".venv/lib/python3.7/site-packages/starlette/middleware/base.py", line 22, in __call__
    return self.app(scope)
  File ".venv/lib/python3.7/site-packages/starlette/exceptions.py", line 49, in __call__
    return self.app(scope)
  File ".venv/lib/python3.7/site-packages/starlette/routing.py", line 587, in __call__
    return route(scope)
  File ".venv/lib/python3.7/site-packages/starlette/routing.py", line 356, in __call__
    return self.app(scope)
  File ".venv/lib/python3.7/site-packages/starlette/applications.py", line 134, in __call__
    return self.error_middleware(scope)
  File ".venv/lib/python3.7/site-packages/starlette/middleware/errors.py", line 88, in __call__
    return self.app(scope)
  File ".venv/lib/python3.7/site-packages/starlette/middleware/sessions.py", line 34, in __call__
    request = Request(scope)
  File ".venv/lib/python3.7/site-packages/starlette/requests.py", line 128, in __init__
    assert scope["type"] == "http"
```

I think the issue is that the session middleware doesn't coimplain when the request type is both http and websocket: https://github.com/encode/starlette/blob/master/starlette/middleware/sessions.py#L32

But right after, creates a Request from the scope when, itself assert only http: https://github.com/encode/starlette/blob/master/starlette/requests.py#L128

### PR DESCRIPTION
**Pull request by taoufik07**
Closes #483

### CODE REVIEW

File: starlette/middleware/sessions.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def __init__(
             await self.app(scope, receive, send)
             return
 
-        request = Request(scope)
+        request = HTTPConnection(scope)
```
Comment (lovelydinosaur) [inline starlette/middleware/sessions.py]: Yup - we absolutely want this, thanks!
Let's change the variable name to `connection = HTTPConnection(scope)` tho.

**Comment by lovelydinosaur** [comment]
Updated your issue description to reference the corresponding issue.

**Comment by taoufik07** [comment]
I didn't notice there was an issue, I was refactoring responder and I found this little typo.

**Comment by lovelydinosaur** [comment]
Great stuff, let's get it in!

**Pull request merged**